include/real/const_precision_iterator.hpp

```diff
@@ -349,7 +349,7 @@ namespace boost {
                            // If the explicit number just reaches the full precision (the end)
                            // then set both boundaries are equals.
                            if (this->_precision + n >= real.digits().size()) {
-                               for(int i = this->_precision; i < (int)real.digits().size(); i++) {
+                               for(precision_t i = this->_precision; i < (int)real.digits().size(); i++) {
```

Comment:
Nice. But you forgot to remove that explicit type conversion at the time of comparison in the loop. That will render this change useless. Please update the conversion part too, then we will merge it. 
The loop should be "for(precision_t i = this->_precision; i < real.digits().size(); ++i)".